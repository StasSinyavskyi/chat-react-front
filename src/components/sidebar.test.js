/* eslint-env jest */
import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import Sidebar from './sidebar';

jest.mock('./new-chat-button', () => () => 'NewChatButton');

// if we have several test we can join them in describe
describe('<Sidebar />', () => {
  const testParam = (
    <Sidebar
      createChat={() => 'createChat'}
      chats={{ active: {}, my: [], all: [] }}
      activeChat={{ }}
      isConnected
    />
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(testParam, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('render correctly', () => {
    const tree = renderer
      .create(testParam)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
