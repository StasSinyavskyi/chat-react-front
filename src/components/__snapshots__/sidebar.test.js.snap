// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sidebar /> render correctly 1`] = `
<div
  className="MuiDrawer-docked-127"
>
  <div
    className="MuiPaper-root-138 MuiPaper-shadow0-140 MuiDrawer-paper-128 Sidebar-drawerPaper-121 MuiDrawer-paperAnchorLeft-129 MuiDrawer-paperAnchorDockedLeft-133"
  >
    <div
      className="Sidebar-drawerHeader-122"
    >
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-165 MuiFormControl-marginNormal-166 MuiFormControl-fullWidth-168"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          className="MuiInput-root-169 MuiInput-fullWidth-177 MuiInput-formControl-170 MuiInput-inkbar-171 MuiInput-underline-175"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          <input
            aria-invalid={false}
            aria-required={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-178"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name={undefined}
            onChange={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder="Search chats..."
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="text"
            value={undefined}
          />
        </div>
      </div>
    </div>
    <hr
      className="MuiDivider-root-184 MuiDivider-default-186"
    />
    <ul
      className="MuiList-root-189 MuiList-padding-190 Sidebar-chatsList-123"
    >
      <h3
        className="MuiTypography-root-193 MuiTypography-subheading-200 Sidebar-noChats-125"
      >
        no chats yet
      </h3>
    </ul>
    NewChatButton
    <div
      className="MuiBottomNavigation-root-217"
    >
      <button
        className="MuiButtonBase-root-225 MuiBottomNavigationAction-root-218"
        disabled={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiBottomNavigationAction-wrapper-221"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-227"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 3c-4.97 0-9 4.03-9 9H1l3.89 3.89.07.14L9 12H6c0-3.87 3.13-7 7-7s7 3.13 7 7-3.13 7-7 7c-1.93 0-3.68-.79-4.94-2.06l-1.42 1.42C8.27 19.99 10.51 21 13 21c4.97 0 9-4.03 9-9s-4.03-9-9-9zm-1 5v5l4.28 2.54.72-1.21-3.5-2.08V8H12z"
            />
          </svg>
          <span
            className="MuiBottomNavigationAction-label-222"
          >
            My Chats
          </span>
        </span>
        <span
          className="MuiTouchRipple-root-234"
        />
      </button>
      <button
        className="MuiButtonBase-root-225 MuiBottomNavigationAction-root-218"
        disabled={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiBottomNavigationAction-wrapper-221"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-227"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 10.9c-.61 0-1.1.49-1.1 1.1s.49 1.1 1.1 1.1c.61 0 1.1-.49 1.1-1.1s-.49-1.1-1.1-1.1zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm2.19 12.19L6 18l3.81-8.19L18 6l-3.81 8.19z"
            />
          </svg>
          <span
            className="MuiBottomNavigationAction-label-222"
          >
            Explore
          </span>
        </span>
        <span
          className="MuiTouchRipple-root-234"
        />
      </button>
    </div>
  </div>
</div>
`;
